src/lib/y2network/hwinfo.rb

```diff
@@ -0,0 +1,75 @@
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+
+module Y2Network
+  # Stores useful (from networking POV) items of hwinfo for an interface
+  # FIXME: decide whether it should read hwinfo (on demand or at once) for a network
+  # device and store only necessary info or just parse provided hash
+  class Hwinfo
+    attr_reader :hwinfo
+
+    def initialize(name:)
+      # FIXME: store only what's needed.
+      @hwinfo = load_hwinfo(name)
+    end
+
+    # Shortcuts for accessing hwinfo items
+    [
+      { name: "dev_name", default: "" },
+      { name: "mac", default: "" },
+      { name: "busid", default: "" },
+      { name: "link", default: false },
+      { name: "driver", default: "" },
+      { name: "drivers", default: [] },
+      { name: "requires", default: [] },
+      { name: "hotplug", default: false },
+      { name: "wl_auth_modes", default: "" },
+      { name: "wl_enc_modes", default: nil },
+      { name: "wl_channels", default: nil },
+      { name: "wl_bitrates", default: nil }
+    ].each do |hwinfo_item|
+      define_method hwinfo_item[:name].downcase do
```

Comment:
why downcase when everything is upcase? or there is keys in hwinfo with upper case? Because this can produce strange looking result like `DevName` -> `devname` where you expect `dev_name`.

Reply:
only readability ... i prefer ```obect.method``` over ```object.METHOD``` or so. And I'm also not sure if there is any exception in naming in the hash provided by libhd